packages/amplify-cli/src/domain/amplify-usageData/UsageData.ts

```diff
@@ -3,29 +3,37 @@ import { Input } from '../input';
 import https from 'https';
 import { UrlWithStringQuery } from 'url';
 import redactInput from './identifiable-input-regex';
-import { UsageDataPayload } from './UsageDataPayload';
+import { ProjectSettings, UsageDataPayload, InputOptions } from './UsageDataPayload';
 import { getUrl } from './getUsageDataUrl';
 import { IUsageData } from './IUsageData';
 import { JSONUtilities } from 'amplify-cli-core';
-
+import _ from 'lodash';
 export class UsageData implements IUsageData {
   sessionUuid: string;
+  accountId: string = '';
   installationUuid: string = '';
   version: string = '';
   input: Input;
+  projectSettings: ProjectSettings;
   url: UrlWithStringQuery;
+  inputOptions: InputOptions;
   requestTimeout: number = 100;
   private static instance: UsageData;
 
   private constructor() {
     this.sessionUuid = uuid.v4();
     this.url = getUrl();
     this.input = new Input([]);
+    this.projectSettings = {};
+    this.inputOptions = {};
   }
 
-  init(installationUuid: string, version: string, input: Input): void {
+  init(installationUuid: string, version: string, input: Input, accountId: string, projectSettings: ProjectSettings): void {
     this.installationUuid = installationUuid;
+    this.accountId = accountId;
+    this.projectSettings = projectSettings;
     this.version = version;
+    this.inputOptions = input.options ? _.pick(input.options as InputOptions, ['sandboxId']) : {};
```

Comment:
is this only storing the sandboxId? if so, I'd just name it that. Also, does this simplify to:
```suggestion
    this.inputOptions = input?.options?.sandboxId || {};
```

Reply:
No we are going to extend this in the future, this is there to add more field if needed in the future